llama_stack/distribution/build_venv.sh

```diff
@@ -49,61 +47,117 @@ ENVNAME=""
 SCRIPT_DIR=$(dirname "$(readlink -f "$0")")
 source "$SCRIPT_DIR/common.sh"
 
+# check if a command is present
+is_present() {
+  command -v "$1" &>/dev/null
+}
+
+# pre-run checks to make sure we can proceed with the installation
+pre_run_checks() {
+  local env_name="$1"
+
+  # checking if uv is installed
+  if ! is_present uv; then
+    echo "uv is not installed, trying to install it."
+    # try to install with pip
+    if ! is_present pip; then
+      echo "pip is not installed, cannot automatically install 'uv'."
+      echo "follow this link to install it:"
+      echo "https://docs.astral.sh/uv/getting-started/installation/"
+      exit 1
+    else
+      pip install uv
+    fi
+  fi
+
+  # checking if an environment with the same name already exists
+  if [ -d "$env_name" ]; then
+    echo "Environment '$env_name' already exists, please remove it."
+    echo "Remove this directory '$env_name'."
+    exit 1
+  fi
+
+  # checking if we are in a venv and if so, deactivate it
```

Comment:
is that needed? I think you could stack venvs?

Reply:
It feels cleaner to deactivate first, but you're right we can stack. What do you want to do?

Reply:
I feel like stacking is the better option.